Modules/Module_Processing/Module_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -254,84 +253,90 @@ function xASL_wrp_ResampleASL(x)
 			else
-				PWI = zeros(size(ASL_im,1), size(ASL_im,2), size(ASL_im,3), blockSize); % preallocate PWI
+				PWI4D = zeros(size(ASL_im,1), size(ASL_im,2), size(ASL_im,3), blockSize); % preallocate PWI
 				for iBlock = 1:blockSize
-					PWI(:,:,:,iBlock) = xASL_stat_MeanNan(ASL_im(:,:,:,iBlock:blockSize:end), 4); % Averaged PWI4D across repetitions
+					PWI4D(:,:,:,iBlock) = xASL_stat_MeanNan(ASL_im(:,:,:,iBlock:blockSize:end), 4); % Averaged PWI4D across repetitions
 				end
 			end
         else % if Hadamard Block size == unique PLDs * number of multi-TEs -> a single repetitions -> we don't want to average
-            PWI = ASL_im;
+            PWI4D = ASL_im;
         end
-
-        % Save PWI4D
-        fprintf('%s\n', PathPWI4D{iSpace});
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI4D{iSpace}, PWI, 32, false);
         
         % Create single PWI for further steps in ASL module
-        PWIsingle = xASL_stat_MeanNan(PWI(:,:,:,1:x.Q.NumberEchoTimes:end),4); % Average across PLDs from each first TE
+        PWI3D = xASL_stat_MeanNan(PWI4D(:,:,:,1:x.Q.NumberEchoTimes:end),4); % Average across PLDs from each first TE
         
-        % Save single PWI
-        fprintf('%s\n', PathPWI{iSpace});
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI{iSpace}, PWIsingle, 32, false);
         
+    % =====================================================================
+    % C) Single- and multi-PLD subtraction
     else
-        % Paired subtraction
-        [ControlIm, LabelIm] = xASL_quant_GetControlLabelOrder(ASL_im);
-        ASL_im = ControlIm - LabelIm;
-
-        % Average PWI - multiPLD
-        if x.modules.asl.bMultiPLD 
-			% Skip every other value in x.Q.Initial_PLD as it was stored for both control and label images 
-			% and we need the PLD vector now for the pairwise subtractions only
-			Initial_PLD_PWI = x.Q.Initial_PLD(1:2:end);
-
-			% The labeling durations also need to be taken into account and unique combinations of PLDs with LDs should be considered
-			if numel(x.Q.Initial_PLD) == numel(x.Q.LabelingDuration)
-				Initial_LabDur_PWI = x.Q.LabelingDuration(1:2:end);
-			else
-				Initial_LabDur_PWI = ones(size(Initial_PLD_PWI)) * x.Q.LabelingDuration;
-			end
+        % Firstly, equalize PLD for single- and multi-PLD
+        nPLD = length(x.Q.Initial_PLD);
+        factorPLD = nVolumes/nPLD;
+        initialPLD = repmat(x.Q.Initial_PLD, [factorPLD 1]);
+        % this vector now has the length of the number of volumes
+        % either all values are identical (in the case of single-PLD)
+        % or a combination of multi-PLD
+        
+        % Secondly, equalize LabDur for single- and multi-PLD
+        nLabDur = length(x.Q.LabelingDuration);
+        factorLabDur = nVolumes/nLabDur;
+        LabDuration = repmat(x.Q.LabelingDuration, [factorLabDur 1]);
         
+        if dim4>1 && ~x.modules.asl.bContainsDeltaM

```

Comment:
As said above - you are forgetting the option with multiple deltaMs. 
If there are multiple deltaMs, you won't create the PWI4Ds and PWI correctly.

Reply:
Fixed in e9bf37aa3f3ef7837a2f0304a17d0d2f1b20342c by only skipping the subtraction if bContainsDeltaM

Reply:
It will still not work - see the other comment.